src/olympia/reviewers/serializers.py

```diff
@@ -171,46 +140,117 @@ def _fetch_entries():
             # cache
             60 * 60 * 24)
 
-        # Fetch and set the sha hash for the currently selected file.
-        sha256 = self._get_hash_for_selected_file(obj)
-        self._entries[self.get_selected_file(obj)]['sha256'] = sha256
-
         return self._entries
 
     def get_mimetype(self, obj):
-        entries = self.get_entries(obj)
-        return entries[self.get_selected_file(obj)]['mimetype']
+        tree = self._get_tree(obj)
+        selected_file = self.get_selected_file(obj)
+        if selected_file in tree:
+            blob_or_tree = tree[selected_file]
+
+            if blob_or_tree.type == pygit2.GIT_OBJ_BLOB:
+                blob = self.git_repo[blob_or_tree.oid]
+                mimetype, mime_category = get_mime_type_for_blob(
+                    tree_or_blob='blob', name=blob_or_tree.name, blob=blob)
+
+                return mimetype
+
+        # If we do not have a blob, return 'application/octet-stream'
+        return 'application/octet-stream'
 
     def get_sha256(self, obj):
-        return self._get_hash_for_selected_file(obj)
+        selected_file = self.get_selected_file(obj)
+        commit = self._get_commit(obj)
+        tree = self._get_tree(obj)
+
+        # Normalize the key as we want to avoid that we exceed max
+        # key lengh because of selected_file.
+        cache_key = make_key(
+            f'reviewers:fileentriesserializer:hashes'
+            f':{commit.hex}:{selected_file}',
+            with_locale=False,
+            normalize=True)
+
+        def _calculate_hash():
+            try:
+                blob_or_tree = tree[selected_file]
+            except KeyError:
+                return None
+
+            if blob_or_tree.type == pygit2.GIT_OBJ_TREE:
+                return None
+
+            blob = self.git_repo[blob_or_tree.oid]
+            return get_sha256(io.BytesIO(memoryview(blob)))
+
+        return cache.get_or_set(cache_key, _calculate_hash, 60 * 60 * 24)
 
     def get_size(self, obj):
-        entries = self.get_entries(obj)
-        return entries[self.get_selected_file(obj)]['size']
+        tree = self._get_tree(obj)
+        selected_file = self.get_selected_file(obj)
+        if selected_file in tree:
+            blob_or_tree = tree[selected_file]
```

Comment:
I saw this code before I think. Should we have a `get_blob_or_tree_for_selected_file` method?

Reply:
Actually, we are only interested in `get_blob_for_selected_file` it seems.